src/bit-systems/network-receive-system.ts

```diff
@@ -71,26 +107,48 @@ export function networkReceiveSystem(world: HubsWorld) {
       const nid = APP.getSid(nidString);
 
       if (world.deletedNids.has(nid)) {
-        // TODO we may need to allow this for reconnects
         console.log(`Received a create message for an entity I've already deleted. Skipping ${nidString}`);
       } else if (world.nid2eid.has(nid)) {
         console.log(`Received create message for entity I already created. Skipping ${nidString}`);
       } else if (!hasPermissionToSpawn(creator, prefabName)) {
         // this should only ever happen if there is a bug or the sender is maliciously modified
-        console.log(`Received create from a user who does not have permission to spawn ${prefabName}`);
+        console.warn(`Received create from a user who does not have permission to spawn ${prefabName}`);
         world.ignoredNids.add(nid); // TODO should we just use deletedNids for this?
       } else {
-        const eid = createNetworkedEntityFromRemote(world, prefabName, initialData, nidString, creator, creator);
+        // TODO: Do we need to delete from ignoredNids?
+        // If permissions have changed, we don't want to permantently ignore messages.
+        // world.ignoredNids.delete(nid);
+
+        const eid = renderAsNetworkedEntity(world, prefabName, initialData, nidString, creator);
         console.log("got create message for", nidString, eid);
+      }
+    }
+  }
+
+  // If we stored updates for newly created entities, queue them for processing
+  enteredNetworkedQuery(world).forEach(eid => {
+    const nid = Networked.id[eid];
+    if (storedUpdates.has(nid)) {
+      console.log("Had stored updates for", APP.getString(nid), storedUpdates.get(nid));
+      const updates = storedUpdates.get(nid)!;
 
-        // If we were hanging onto updates for this nid we can now apply them. And they should be processed before other updates.
-        if (storedUpdates.has(nid)) {
-          console.log("had pending updates for", nidString, storedUpdates.get(nid));
-          Array.prototype.unshift.apply(message.updates, storedUpdates.get(nid));
-          storedUpdates.delete(nid);
+      for (let i = 0; i < updates.length; i++) {
+        const update = updates[i];
+        if (partedClientIds.has(APP.getSid(update.owner))) {
+          console.log("Rewriting update message from client who left.", JSON.stringify(update));
+          update.owner = NAF.clientId;
+          update.lastOwnerTime = update.timestamp;
         }
       }
+
+      // Process the stored message before other updates
+      pendingMessages.unshift({ creates: [], updates, deletes: [] });
+      storedUpdates.delete(nid);
     }
+  });
+
+  for (let i = 0; i < pendingMessages.length; i++) {
```

Comment:
I suppose I could go either way, but prefer to split it up like we've done:

```
// Handle parting clients
// Delete entities
// Create entities
// Update entities (include any stored updates)
// Takeover abandoned entities
```

Rather than:

```
// Handle parting clients
// For each message:
//   delete
//   create
//   update (including any stored updates)
// Takeover abandoned entities
```

It seems desirable that it's clear to readers that we don't have to process all of a given client's messages in order before moving on, and that instead the receive system can be focused more broadly on, "How am I going to manipulate the game state next?"

It's not a perfect analogy, but it feels similar to how we've written the ECS systems to be more about, "What's the next transformation to component datas (across all entities) do we need to make?" rather than, "What's the next object/entity we need to update?"

"What's the batch of entities I need to delete?", not "What are all the things I need to do in response to this message".